Improve resolution of relative includes

hanssv opened the pull request:
Fixes #488

This PR is supported by the Æternity Crypto Foundation

radrow commented on CHANGELOG.md:
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 ### Added
 ### Changed
+- Improve how includes with relative paths are resolved during parsing/compilation.
```
I think it should be clearer on how they are improved. Also, this change should be documented - for example it is not clear which path takes priority if there is a file available relative to the current file and another one relative to cwd

radrow replied:
Test needed

ghallak commented on CHANGELOG.md:
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 ### Added
 ### Changed
+- Improve how includes with relative paths are resolved during parsing/compilation.
```
Something like this explains the change in a better way.

```suggestion
- Resolve include paths relative to the including file path when the resolution relative to the root file path fails.
```

hanssv replied:
Reformulated.

hanssv commented:
Added documentation + reformulated CHANGELOG.

Reworked the solution a bit, by always tracking the current src_dir in `Options`. This is slightly more verbose but a lot cleaner I think.

And added the missing test case from #488
